File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for (int i = 0; i < s.length(); i++)
        {
            if (s[i] == '(' || s[i] == '{' || s[i] == '[')
                st.push(s[i]);
            else
            {
                if (s[i] == ')')
                {
                    if (st.size() > 0 && st.top() == '(')
                        st.pop();
                    else
                        return (false);
                }
                else if (st.size() > 0 && s[i] == '}')
                {
                    if (st.top() == '{')
                        st.pop();
                    else
                        return (false);
                }
                else
                {
                    if (st.size() > 0 && st.top() == '[')
                        st.pop();
                    else
                        return (false);
                }
            }
        }
        return (st.size() == 0);
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack = [];
        for char in s:
            if (char == ']'):
                if (len(stack) == 0 or stack.pop() != '['):
                    return (False);
            elif (char == '}'):
                if (len(stack) == 0 or stack.pop() != '{'):
                    return (False);
            elif (char == ')'):
                if (len(stack) == 0 or stack.pop() != '('):
                    return (False);
            else:
                stack.append(char);
        return (len(stack) == 0);